clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -401,10 +401,14 @@ public boolean hasUnsent() {
         return batches;
     }
 
+    private Deque<RecordBatch> getDeque(TopicPartition tp) {
```

Comment:
TBH, I do not have any better suggestion as well. Maybe it is OK to have it the way it is.